test/CMakeLists.txt

```diff
@@ -78,6 +78,15 @@ add_fmt_test(printf-test)
 add_fmt_test(ranges-test ranges-odr-test.cc)
 add_fmt_test(scan-test)
 add_fmt_test(std-test)
+try_compile(compile_result_unused
+            ${CMAKE_CURRENT_BINARY_DIR}
+            SOURCES ${CMAKE_CURRENT_LIST_DIR}/detect-stdfs.cc
+            CMAKE_FLAGS CMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}
+            OUTPUT_VARIABLE RAWOUTPUT)
+string(REGEX REPLACE ".*libfound \"([^\"]*)\".*" "\\1" STDLIBFS "${RAWOUTPUT}")
+if (STDLIBFS)
+  target_link_libraries(std-test ${STDLIBFS})
+endif ()
```

Comment:
Is it possible to simplify this?

Reply:
The FS-library name is determined by the version of the C++ Standard Library.
CMake only determine the compiler version.
Clang supports ``libstdc++`` and ``libcxx``.
Intel C++ supports ``libstdc++`` on Linux, ``libcxx`` on macOS, and Microsoft STL on Windows.
Therefore, the only way to determine which standard library is actually used is by compiling the code.

The code I wrote is the easiest way I know.

Reply:
Can this information be derived from `CMAKE_CXX_STANDARD_LIBRARIES` or `CMAKE_CXX_IMPLICIT_LINK_LIBRARIES`?

Reply:
The variable ``CMAKE_CXX_IMPLICIT_LINK_LIBRARIES`` allows to detect the standard library name, but not its version.

Example:
```
GCC-8:
    CMAKE_CXX_IMPLICIT_LINK_LIBRARIES: stdc++;m;gcc_s;gcc;c;gcc_s;gcc

GCC-9:
    CMAKE_CXX_IMPLICIT_LINK_LIBRARIES: stdc++;m;gcc_s;gcc;c;gcc_s;gcc

Clang 13 + libstdc++ (from GCC-11):
    CMAKE_CXX_IMPLICIT_LINK_LIBRARIES: stdc++;m;gcc_s;gcc;c;gcc_s;gcc

Clang 13 + libc++:
    CMAKE_CXX_IMPLICIT_LINK_LIBRARIES: c++;m;gcc_s;gcc;c;gcc_s;gcc
```

``CMAKE_CXX_STANDARD_LIBRARIES`` - CMake 3.6+ and empty in most configurations